### DISCUSSION TITLE
build leptonica x64-linux static with -fPIC

### DISCUSSION THREAD

**Discussion by blemidon** [microsoft/vcpkg#26192] [Q&A] [answered]
Hi!
I'm using leptonica&tesseract on linux. It builds fine, but when I link it into my dynamic lib, I get the error:
error: .../vcpkg/installed/x64-linux/lib/libleptonica.a(jp2kio.c.o): requires dynamic R_X86_64_PC32 reloc against 'opj_stream_create' which may overflow at runtime; recompile with -fPIC

I modified both leptonica & tesseract portfiles with
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
       ...
        -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE
)
but the error remains.
vcpkg version is 2022.06.15

I'm on OpenSuse Leap 15.4, g++ (SUSE Linux) 11.3.0, GNU gold (GNU Binutils; SUSE Linux Enterprise 15 2.37.20211103-150100.7.37) 1.16

Thanks in advance:
imrei

**Comment by dg0yt** [comment]
In general, vcpkg already builds PIC also for static triplets (x64-linux). Did you actually update all installed ports, including `openjpeg`?

**Comment by blemidon** [comment, marked as answer]
I've tried it with 2022.07.25 and it works without any modification,
thanks
